shared_model/bindings/model_crypto.cpp

```diff
@@ -21,8 +21,31 @@
 namespace shared_model {
   namespace bindings {
     crypto::Keypair ModelCrypto::generateKeypair() {
-      // TODO: 06.12.2017 luckychess create keypair generation interface IR-684
       return crypto::CryptoProviderEd25519Sha3::generateKeypair();
     }
+
+    crypto::Keypair ModelCrypto::generateKeypair(const std::string &seed) {
+      return crypto::CryptoProviderEd25519Sha3::generateKeypair(
+          crypto::Seed(seed));
+    }
+
+    crypto::Keypair ModelCrypto::convertFromExisting(
+        const std::string &publicKey, const std::string &privateKey) {
+      crypto::Keypair keypair((crypto::Keypair::PublicKeyType(
```

Comment:
redundant bracket?

Reply:
Surprisingly, but no :) Without double brackets it will not compile, say hello to https://en.wikipedia.org/wiki/Most_vexing_parse

Reply:
holy mango of charango
this lang is awful D: